x-pack/plugins/infra/public/metrics_overview_fetchers.ts

```diff
@@ -0,0 +1,161 @@
+  nodes: SnapshotNode[],
+  combinator: (values: number[]) => number
+) => {
+  const values = nodes.reduce((acc, node) => {
+    const snapshotMetric = node.metrics.find((m) => m.name === metric);
+    if (snapshotMetric?.value != null && isFinite(snapshotMetric.value)) {
+      acc.push(snapshotMetric.value);
+    }
+    return acc;
+  }, [] as number[]);
+  return combinator(values);
+};
+
+interface CombinedRow {
+  values: number[];
+  timestamp: number;
+}
+
+export const combineNodeTimeseriesBy = (
+  metric: SnapshotMetricType,
+  nodes: SnapshotNode[],
+  combinator: (values: number[]) => number
+) => {
+  const combinedTimeseries = nodes.reduce((acc, node) => {
+    const snapshotMetric = node.metrics.find((m) => m.name === metric);
+    if (snapshotMetric && snapshotMetric.timeseries) {
+      snapshotMetric.timeseries.rows.forEach((row) => {
+        const combinedRow = acc.find((r) => r.timestamp === row.timestamp);
+        if (combinedRow) {
+          combinedRow.values.push(isNumber(row.metric_0) ? row.metric_0 : 0);
+        } else {
+          acc.push({
+            timestamp: row.timestamp,
+            values: [isNumber(row.metric_0) ? row.metric_0 : 0],
+          });
+        }
+      });
+    }
+    return acc;
+  }, [] as CombinedRow[]);
+  return combinedTimeseries.map((row) => ({ x: row.timestamp, y: combinator(row.values) }));
+};
+
+export const createMetricsFetchData = (
+  getStartServices: InfraClientCoreSetup['getStartServices']
+) => async ({
+  startTime,
+  endTime,
+  bucketSize,
+}: FetchDataParams): Promise<MetricsFetchDataResponse> => {
+  const [coreServices] = await getStartServices();
+  const { http } = coreServices;
+  const snapshotRequest: SnapshotRequest = {
+    sourceId: 'default',
+    metrics: ['cpu', 'memory', 'rx', 'tx'].map((type) => ({ type })) as SnapshotMetricInput[],
+    groupBy: [],
+    nodeType: 'host',
+    timerange: {
+      from: moment(startTime).valueOf(),
+      to: moment(endTime).valueOf(),
+      interval: bucketSize,
+      forceInterval: true,
+      ignoreLookback: true,
+    },
+  };
+
+  const results = await http.post<SnapshotNodeResponse>('/api/metrics/snapshot', {
+    body: JSON.stringify(snapshotRequest),
+  });
+
+  const inboundLabel = i18n.translate('xpack.infra.observabilityHomepage.metrics.rxLabel', {
+    defaultMessage: 'Inbound traffic',
+  });
+
+  const outboundLabel = i18n.translate('xpack.infra.observabilityHomepage.metrics.txLabel', {
+    defaultMessage: 'Outbound traffic',
+  });
+
+  return {
+    title: i18n.translate('xpack.infra.observabilityHomepage.metrics.title', {
+      defaultMessage: 'Metrics',
+    }),
+    appLink: '../app/metrics',
+    stats: {
+      hosts: {
+        type: 'number',
+        label: i18n.translate('xpack.infra.observabilityHomepage.metrics.hostsLabel', {
+          defaultMessage: 'Hosts',
+        }),
+        value: results.nodes.length,
+      },
+      cpu: {
+        type: 'number',
```

Comment:
@formgeist can correct me if I'm wrong, but based on the design, `CPU`, `Memory`, `Disk Usage` must be in `percent`, and `InboundTraffic` and `OutboundTraffic` must be `bytesPerSecond`. The type property accepts `'number' | 'percent' | 'bytesPerSecond'`

<img width="509" alt="Screenshot 2020-07-02 at 08 54 04" src="https://user-images.githubusercontent.com/55978943/86326370-e50dbb80-bc41-11ea-9e69-0dccc7013019.png">

Reply:
Indeed - I grabbed the metric examples from 
<img width="1351" alt="Screenshot 2020-07-02 at 09 38 53" src="https://user-images.githubusercontent.com/4104278/86330495-0ec9e100-bc48-11ea-9f88-f27f2d02a939.png">

Reply:
Does the `type` field define the formatter you're going to use? For example, the fields that are listed as percents, do you want the. numbers to be `0.01` or `1` for something that is `1%`?

Reply:
1 means 100%.

Reply:
Just wanted to make sure we are all on the same page. I wonder instead of `type` that should be called `format`?

Reply:
that sounds good too, but I'd rather stick with type for now. If that's fine for you too.

Reply:
Works for me if it works for you :D